src/query/api/v1/middleware/rewrite.go

```diff
@@ -263,3 +337,44 @@ func maybeUpdateLookback(
 	}
 	return false, lookback
 }
+
+type fakeQueryable struct {
+	engine              *promql.Engine
+	instant             bool
+	calculatedStartTime time.Time
+	calculatedEndTime   time.Time
+}
+
+func (f *fakeQueryable) Querier(ctx context.Context, mint, maxt int64) (promstorage.Querier, error) {
+	f.calculatedStartTime = xtime.FromUnixMillis(mint)
+	f.calculatedEndTime = xtime.FromUnixMillis(maxt)
+	// fail here to cause prometheus to give up on query execution
+	return nil, errIgnorableQuerierError
+}
+
+func (f *fakeQueryable) calculateQueryBounds(
+	q string,
+	start time.Time,
+	end time.Time,
+	step time.Duration,
+) (err error) {
+	var query promql.Query
+	if f.instant {
+		// startTime and endTime are the same for instant queries
+		query, err = f.engine.NewInstantQuery(f, q, start)
+	} else {
+		query, err = f.engine.NewRangeQuery(f, q, start, end, step)
+	}
+	if err != nil {
+		return err
+	}
+	// The result returned by Exec will be an error, but that's expected
+	if res := query.Exec(context.Background()); errors.Is(res.Err, errIgnorableQuerierError) {
```

Comment:
Don't we want to _not_ return an error if the value is `errIgnorableQuerierError`?